### DISCUSSION TITLE
Can't load Q16HDRI.bundle for module Image::Magick::Q16HDRI Mac OS

### DISCUSSION THREAD

**Discussion by schliden** [ImageMagick/ImageMagick#3058] [Help]
I have a script that requires ImageMagick and points me to https://www.imagemagick.org/script/perl-magick.php for installation.

I installed PerlMagick from the ImageMagick source distribution by following instructions for manual installation.

When I try to run the script I get the following error:

```
Can't load '/Library/Perl/5.18/darwin-thread-multi-2level/auto/Image/Magick/Q16HDRI/Q16HDRI.bundle' for module Image::Magick::Q16HDRI: dlopen(/Library/Perl/5.18/darwin-thread-multi-2level/auto/Image/Magick/Q16HDRI/Q16HDRI.bundle, 1): Library not loaded: /usr/local/opt/jpeg/lib/libjpeg.9.dylib
  Referenced from: /usr/local/opt/libtiff/lib/libtiff.5.dylib
  Reason: Incompatible library version: libtiff.5.dylib requires version 14.0.0 or later, but libjpeg.9.dylib provides version 13.0.0 at /System/Library/Perl/5.18/darwin-thread-multi-2level/DynaLoader.pm line 194.
 at /System/Library/Perl/5.18/parent.pm line 20.

Compilation failed in require at /System/Library/Perl/5.18/parent.pm line 20.
BEGIN failed--compilation aborted at /Library/Perl/5.18/darwin-thread-multi-2level/Image/Magick.pm line 22.
Compilation failed in require at /Users/username/bin/smlcli/smlcli.pl line 16.
BEGIN failed--compilation aborted at /Users/username/bin/smlcli/smlcli.pl line 16.
&Image::Magick::Q16HDRI::constant not defined. The required ImageMagick libraries are not installed or not installed properly.
END failed--call queue aborted at /Users/username/bin/smlcli/smlcli.pl line 16.
```

Any advice to fix this issue ?

Thanks

**Comment by urban-warrior** [comment]
The message suggests that your ImageMagick library and your PerlMagick build were against two different versions of the TIFF and JPEG delegate libraries.  Check your system for multiple instances of these libraries (e.g. libtiff, libjpeg) and ensure that both ImageMagick and PerlMagick build against the same versions.  Another option is to install ImageMagick and PerlMagick from the same source build.  Add `--with-perl` to your `configure` command-line and both ImageMagick and PerlMagick will build together and ensure they remain compatible.

**Reply by schliden** [reply]
Thanks for your advice.

I've looked a little deeper and realised that 

/usr/local/opt/jpeg/  and  /usr/local/opt/libtiff/  are symlink'd into /usr/local/Cellar/  ... Home Brew.

So not sure if that might be contributing to the issue and if solutions same with Home Brew involved.
